### ISSUE TITLE
Combining multiple models with multiple outputs.

### ISSUE THREAD

**Issue by sronast** [keras-team/keras#14279]
I have 2 models A and B.
A = tensorflow.keras.Model(inputs = i1, outputs=[o1, o2])

B = tensorflow.keras.Model(inputs = i2, outputs=[o3, o4])

How can I combine above two models so that the final model will have 2 input layers and 4 outputs layers.? Also how can I train the final model?

**Comment by pmeier-tiplu** [comment]
have you tried wrapping it in another keras.model?

```python
# given
A = tensorflow.keras.Model(inputs = i1, outputs=[o1, o2])
B = tensorflow.keras.Model(inputs = i2, outputs=[o3, o4])

# create new model
C = tensorflow.keras.Model(inputs = A.inputs + B.inputs, outputs= A.outputs + B.outputs)
```

**Comment by sronast** [comment]
> have you tried wrapping it in another keras.model?
> 
> ```python
> # given
> A = tensorflow.keras.Model(inputs = i1, outputs=[o1, o2])
> B = tensorflow.keras.Model(inputs = i2, outputs=[o3, o4])
> 
> # create new model
> C = tensorflow.keras.Model(inputs = A.inputs + B.inputs, outputs= A.outputs + B.outputs)
> ```

I tried combining like this:

o1, o2 = A(i1)
o3, o4 = B(i2)

C = tensorflow.keras.Model(inputs =[i1, i2], outputs= [[o1,o2], [o3,o4]])

C.compile(loss=['custom_loss_A', 'custom_loss_B'],...)

And while training the model C, I tried:
C.train_on_batch([x_i1, x_i2], [[y_o1, y_o2],[y_o3, y_o4]])


I defined  custom loss functions for both models A  and B both of which are supposed to take two predicted outputs and two true output values.   
For example, loss function for A should take ([o1, o2] , [y_o1, y_o2]) for calculating loss. But while training the model this loss function is taking only (o1, y_o1).

Could you help me solving this problem?
